### ISSUE TITLE
Natural numbers depend on chaos-pp

### ISSUE THREAD

**Issue by raedwulf** [rofl0r/order-pp#2]
Order-pp is meant to be standalone but https://github.com/rofl0r/order-pp/blob/master/inc/order/prelude/nat.h uses chaos-pp.

I suppose we need to reimplement those in order-pp to rectify it.

**Comment by rofl0r** [comment]
yeah, feel free to do so (or to just copy them over and rename them).

**Comment by raedwulf** [comment]
I was looking at either method but it seems that those dependencies pull in the rest of chaos-pp... I wish I knew more about the internals so I could port it!

**Comment by raedwulf** [comment]
It looks like the add operator is implemented in order-pp but the others are not.. so this is a good starting point for reimplementing it in order!
